modules/document_repository/jsx/docIndex.js

```diff
@@ -0,0 +1,196 @@
+import {Tabs, TabPane} from 'Tabs';
+import DocUploadForm from './uploadForm';
+import DocCategoryForm from './categoryForm';
+import Loader from 'Loader';
+import FilterableDataTable from 'FilterableDataTable';
+
+class DocIndex extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isLoaded: false,
+      data: {},
+      error: false,
+      newCategory: false,
+    };
+
+    // Bind component instance to custom methods
+    this.fetchData = this.fetchData.bind(this);
+    this.formatColumn = this.formatColumn.bind(this);
+    this.newCategoryState = this.newCategoryState.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
+  }
+  fetchData() {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data}))
+      .then(()=>this.setState({newCategory: false}))
+      .catch((error) => {
+        this.setState({error: true});
+    });
+  }
+
+  newCategoryState() {
+    this.setState({newCategory: true});
+  }
+/**
+ * Modify behaviour of specified column cells in the Data Table component
+ * @param {string} column - column name
+ * @param {string} cell - cell content
+ * @param {arrray} row - array of cell contents for a specific row
+ * @return {*} a formated table cell for a given column
+ */
+ formatColumn(column, cell, row) {
+ let result = <td>{cell}</td>;
+   switch (column) {
+   case 'File Name':
+   let downloadURL = loris.BaseURL + '/document_repository/ajax/GetFile.php?File=' + encodeURIComponent(row['File Name']);
```

Comment:
This url doesn't look right.. shouldn't it be`loris.BaseURL + '/document_repository/Files/' + encodeURIComponent(row['File Name']`? (Based on the description of the files class..)